### DISCUSSION TITLE
Smoke Levels

### DISCUSSION THREAD

**Discussion by weberbox** [nebhead/PiFire#133] [Ideas]
Maybe I am jumping the gun since I have not even played around with PWM that much yet but what are your thoughts on the Camp Chef smoke levels and how they work. 

> Your pellet grill feeds pellets at a known rate to maintain the temperature you've set. (This is done by the auger motor cycling ON and OFF at a calculated, PID adjusted, engineered rate.) To increase the amount of smoke generated, we increase the auger's OFF time. This allows more pellets to be burned, decreasing the flame and ember temperature. So now, when new pellets are fed into the burn cup it'll take longer for them to ignite, resulting in an increase of smoke. Auger FEED times are also increased so that there is, on average, enough fuel to maintain your set temperature. These fluctuations in temperature and ignition are what give us "tasty" smoke. Temperature fluctuations are common and are completely normal. In fact, because your controller now has PID, you're getting the industry's most precise grill temperature (+/- 5F.) As your Smoke Number increases, your smoke output increases. As your Smoke Number decreases the temperatures will hold tighter.

I have noticed that even with slowing the fan down or turning it off and on the best smoke is when the pellet flame extinguishes and they actually start to smother.  This does not always happen even when the fan is turning off and on. 

I almost want to buy one of the controllers to see how they setup the pid. 

I was also thinking about taking a stab at creating a screen very similar to theirs as it could support 3 food probes, 1 grill probe and have a spot for the icons like ignitor etc down in the lower right where their fourth probe is.  Also the bottom line could show count down timers, time elapsed since started (I think that would be a good addition) as well as pellet level etc. 

![image](https://user-images.githubusercontent.com/313328/195741570-7f308578-6fd7-4137-af75-7e1eb45b212b.png)

**Comment by nebhead** [comment]
> Maybe I am jumping the gun since I have not even played around with PWM that much yet but what are your thoughts on the Camp Chef smoke levels and how they work.

Very interesting.  So my take is that basically they are bumping up the Auger On time and Auger Off time by a certain amount to ensure they are feeding more pellets into the firepot, and also giving them more time to burn between cycles.  It's an interesting idea and maybe worth playing around with.  It goes along with the theory of priming your pot for a big initial smoke.  More pellets in the pot means more smoke output, at least in theory.  

Apparently Green Mountain Grills has moved away from PID to more of a spreadsheet system, with hard coded cycle times, for better smoke.  I think this is also an interesting idea, however, for this project it would be difficult because everyone has different cooking chambers with different sizes.  Regardless, would be interesting to be able to load a profile for a specific manufacturer and use that instead.  

> I have noticed that even with slowing the fan down or turning it off and on the best smoke is when the pellet flame extinguishes and they actually start to smother. This does not always happen even when the fan is turning off and on.

Yes, I think that's the magic of the fan cycling, is that it gives it a chance to almost smother then come back to life.  This temporarily generates just a little more smoke from the pellets.  

> I almost want to buy one of the controllers to see how they setup the pid.

What we need, is to find someone on Reddit or in the forums that would be willing to help reverse engineer it. :P 
 
> I was also thinking about taking a stab at creating a screen very similar to theirs as it could support 3 food probes, 1 grill probe and have a spot for the icons like ignitor etc down in the lower right where their fourth probe is. Also the bottom line could show count down timers, time elapsed since started (I think that would be a good addition) as well as pellet level etc.

That would great!  It's a very clean looking layout for sure.  Maybe create a new base module (you could even use the existing gauge code if you want), and give it a go!
